### ISSUE TITLE
使用develop版本的paddledetection边训练边评估mask-rcnn,无评估结果

### ISSUE THREAD

**Issue by ResonLee** [PaddlePaddle/PaddleDetection#3671]
cuda：10.1
paddlepaddle：develop版本
paddledetection：develop版本
显卡：2080Ti*2
系统：ubuntu 16.04
使用paddle mask-Rcnn训练自己的数据集，（数据集没问题，用solov2训练过的）模型使用mask_rcnn_r50_fpn_1x_coco.yml，使用对应的预训练权重文件mask_rcnn_r50_fpn_1x_coco.pdparams，训练配置文件如下：
epoch: 36
LearningRate:
  base_lr: 0.0025
  schedulers:
  - !PiecewiseDecay
    gamma: 0.1
    milestones: [24, 32]
  - !LinearWarmup
    start_factor: 0.001
    steps: 2000
OptimizerBuilder:
  optimizer:
    momentum: 0.9
    type: Momentum
  regularizer:
    factor: 0.0001
    type: L2
 训练过程的评估：
[07/12 15:00:51] ppdet.engine INFO: Eval iter: 0
[07/12 15:01:01] ppdet.engine INFO: Eval iter: 100
[07/12 15:01:11] ppdet.engine INFO: Eval iter: 200
[07/12 15:01:21] ppdet.engine INFO: Eval iter: 300
[07/12 15:01:27] ppdet.engine INFO: Total sample number: 370, averge FPS: 10.129556143074254
[07/12 15:01:27] **ppdet.engine WARNING: Evaluation results empty, this may be due to training iterations being too few or not loading the correct weights.**
使用训练完后的权重模型来测试图像，也没检测出任何目标，请问是怎么回事啊？

**Comment by nemonameless** [comment]
是训练了几个epoch测的？可能是训练早期效果太低。建议可以用训好的coco模型去finetune，效果会比从头开始训要好很多，一开始精度也很高了。

**Comment by JackLee-Python** [comment]
训练36个epoch, 训练效果低的话也不至于最后一轮也是这样吧，而且训练完后检测自己的数据，任何目标都检测不出来啊
在说mask-rcnn的mask_rcnn_r50_fpn_1x_coco网络性能应该也可以吧，半个月前用还是能训练出来的，但是这次是任何东西都训练不出来

**Comment by ResonLee** [comment]
@nemonameless 训练36个epoch, 训练效果低的话也不至于最后一轮也是这样吧，而且训练完后检测自己的数据，任何目标都检测不出来啊
说mask-rcnn的mask_rcnn_r50_fpn_1x_coco网络性能应该也可以吧，半个月前用还是能训练出来的，但是这次是任何东西都训练不出来

**Comment by nemonameless** [comment]
> @nemonameless 训练36个epoch, 训练效果低的话也不至于最后一轮也是这样吧，而且训练完后检测自己的数据，任何目标都检测不出来啊
> 说mask-rcnn的mask_rcnn_r50_fpn_1x_coco网络性能应该也可以吧，半个月前用还是能训练出来的，但是这次是任何东西都训练不出来

训练log最后方便截图看看吗?半个月前训代码是同一个分支同一份吗？

**Comment by ResonLee** [comment]
@nemonameless 
![image](https://user-images.githubusercontent.com/54474938/125388858-da4c4180-e3d2-11eb-8234-9efee4333477.png)
最后训练训练日志大概都是这样，今天又试了两次，还是这样，这次训练到17轮就让它停下来了。
同一个develop分支，这次训练的环境是重新配的。

**Comment by nemonameless** [comment]
训练过程不正常，几个loss是什么时候开始变成nan的？在某个loss变nan之前的那个epoch可以测下看是多少。loss为nan的原因很多不易排查，急用的话建议可以用训好的coco模型去finetune。

**Comment by ResonLee** [comment]
@nemonameless 从第一个epoch开始，就出现了nan,如下图， 我就是加载官方的预训练模型做finetune训练的，训练命令为“CUDA_VISIBLE_DEVICES=6,7 python tools/train.py -c configs/mask_rcnn/mask_rcnn_r50_fpn_1x_coco.yml --use_vdl=true --vdl_log_dir=vdl_log_dir/scalar -o pretrain_weights=weights/mask_rcnn_r50_fpn_1x_coco.pdparams --eval”
![image](https://user-images.githubusercontent.com/54474938/125398533-08854d80-e3e2-11eb-859f-42863825c8d2.png)

**Comment by heavengate** [comment]
看你训练过程中出nan了，看训练命令是两卡跑的，有可能是现在配置的针对8卡的学习率对于两卡来说过高了，可以参考FAQ调整一下学习率试一下
https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.1/docs/tutorials/FAQ.md

**Comment by ResonLee** [comment]
@nemonameless 训练前学习率按这个改了，2个GPU设置学习率0.0025，训练batch_szie为4

**Comment by heavengate** [comment]
有修改其他的配置么，看你log应该就是训练过程中出nan了，模型训飞了没有收敛，或者可以试下用我们提供的默认配置和coco数据集能否正常训练，如果可以训练的话应该是配置或者数据集的问题，可以排查一下你的数据集里有没有不合法的越界的标注之类的
